code-challenges.test.js

```diff
@@ -15,46 +15,164 @@
+//example say input or count = 5, we expect => [1, 1, 2, 3, 5]
+//fibSeq(5) is added to the call stack, and returns array = fibSeq(4) 
+//fibSeq(4) is added to the call stack, and returns array = fibSeq(3)
+//fibSeq(3) is added to the call stack, and returns array = fibSeq(2)
+//fibSeq(2) is added to the call stack, since count = 2, we execute else if statement and returns array = fibSeq(1)
+//fibSeq(1) is added to the call stack, since count = 2, we execute else if statement and returns array = fibSeq(0)
+//this is where we hit the base case and where we find out fibSeq(0) returns []
+//we now work last in first out on the call stack
+//at fibSeq(1) we know array = [] so we push 1 into the array, it returns [1]
+//at fibSeq(2) array = [1] so we push another 1 into the array, it returns [1, 1]
+//at fibSeq(3) array = [1, 1] so we push the addition of (array[0] + array[1]) OR (1+1), returns [1, 1, 2]
+//at fibSeq(4) array = [1, 1, 2] we push (array[1] + array[2]) OR (1+2), returns [1, 1, 2, 3]
+//at fibSeq(5) array = [1, 1, 2, 3] we push (array[2] + array[3]) OR (2+3), returns [1, 1, 2, 3, 5]
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+//JEST TEST to test an onlyOdd function that filters out odd nums in an array and sorts them least to great
+describe("tests onlyOdd function", ()=>{
+    it("takes in array and rtrns only odd nums from least to greatest", () =>{
+
+        var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        // Expected output: [-9, 7, 9, 199]
 
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
+        var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+        // Expected output: [-823, 7, 23]
+
+        expect(onlyOdd(fullArr1)).toEqual([-9, 7, 9, 199]);
+        expect(onlyOdd(fullArr2)).toEqual([-823, 7, 23]);
+    })
+})
+//jest test fails because "ReferenceError: onlyOdd is not defined"
 
 
 // b) Create the function that makes the test pass.
 
+//create fn onlyOdd that takes array param
+const onlyOdd = (array) =>{
+    //filters array to return only odd values AND only with typeof "number"
+    let oddsArray = array.filter(value =>{ 
+        return value % 2 !== 0 && typeof value === "number"
+    });
+
+    //returns the sorted oddsArray from smallest to largest where if a-b < 0 sort a before b
+    return oddsArray.sort((a, b) => a-b);
+}
+//failed test because it seems filter does type coercion regardless of the strictly not equals and because sort fn will sort based on strings if no comparison is identified. Passed test after fixing it. Can't really think of a way this might need refactoring.
```

Comment:
The only thing here would be appending the dot notation into one line. 
```javascript 
return array.filter(value => return value % 2 !== 0 && typeof value === "number").sort((a, b) => a-b);
```

Reply:
Thanks! I appreciate this feedback.